datasets/restaurant.py

```diff
@@ -0,0 +1,232 @@
+import random
+from typing import List, Tuple, Iterator, Callable
+
+from datetime import datetime, timedelta
+from dataclasses import dataclass, field
+from dataset_interfaces.interface import DatasetInterface, TestExample, TestAction, SendMessageAction, WaitAction
+
+from utils.openai import ask_llm, LLMContext, make_user_message
+from goodai.helpers.json_helper import sanitize_and_parse_json
+
+
+@dataclass
+class RestaurantExample(TestExample):
+    cost_callback: Callable[[float], None] = None
```

Comment:
The cost_callback is a member of the `DatasetInterface` class so you might be able to use that?

Reply:
Yes. `DynamicExample` objects have it because they might perform calls to gpt-3 to evaluate things or guide the script. The cost is tiny, but this way it's accounted for.